stories/patterns/filters/add-filter/examples/docs/add-filter.tsx

```diff
@@ -0,0 +1,309 @@
+        <Button.Addon>
+          <SearchM />
+        </Button.Addon>
+      </Button>
+    </Flex>
+  );
+};
+
+type KeywordDataItem = {
+  value: string;
+  displayValue: string;
+} | null;
+
+type KeywordProps = {
+  value: KeywordDataItem;
+  onChange: (v: KeywordDataItem) => void;
+};
+const Keywords = ({ value, onChange }: KeywordProps) => {
+  const [textAreaValue, setTextAreaValue] = React.useState(value?.value ?? '');
+
+  const applyFilters = () => {
+    if (!textAreaValue) {
+      return;
+    }
+    const countLine = textAreaValue.match(/\n/g) || [];
+    const value = textAreaValue;
+    const displayValue = String(countLine.length || (textAreaValue && 1));
+
+    onChange({
+      value,
+      displayValue,
+    });
+  };
+
+  return (
+    <>
+      <Text tag='label' htmlFor='textarea' size={200} color='text-primary'>
+        Enter keywords separated by commas or one per line. For exact matches, enter your keyword
+        with square brackets around it.
+      </Text>
+      <RadioGroup my={4} defaultValue='1' direction='row'>
+        <Radio>
+          <Radio.Value value='1' />
+          <Radio.Text>All keywords</Radio.Text>
+        </Radio>
+        <Radio ml={6}>
+          <Radio.Value value='2' />
+          <Radio.Text>Any keywords</Radio.Text>
+        </Radio>
+      </RadioGroup>
+      <Textarea value={textAreaValue} onChange={setTextAreaValue} h={132} id='textarea' />
+      <Flex mt={5}>
+        <Button use='primary' theme='info' onClick={applyFilters}>
+          Apply
+        </Button>
+        <Button
+          ml={2}
+          onClick={() => {
+            onChange(null);
+          }}
+        >
+          Clear all
+        </Button>
+      </Flex>
+    </>
+  );
+};
+
+type FilterData = {
+  name: string;
+  fullname: string;
+  size: string;
+  position: string;
+  device: string;
+  keywords: KeywordDataItem | null;
+};
+const defaultFilterData = {
+  name: '',
+  fullname: '',
+  size: '',
+  keywords: null,
+  device: '',
+  position: '',
+};
+const AddFilterPatternExample = () => {
+  const [filterData, setFilterData] = React.useState<FilterData>(() => defaultFilterData);
+
+  const clearField = React.useCallback(
+    (name: keyof FilterData) => {
+      const valueType = typeof filterData[name];
+      const tempData = { ...filterData, [name]: valueType === 'string' ? '' : null };
+      setFilterData(tempData);
+    },
+    [filterData],
+  );
+
+  return (
+    <AddFilterPattern
+      onClearAll={() => {
+        setFilterData(defaultFilterData);
+      }}
+      gap={2}
+      flexWrap
+    >
+      <AddFilterPattern.Search alwaysVisible={true} name='name' displayName='Name'>
+        <SearchFilterInput
+          placeholder={'Filter by name'}
+          onChange={(v: string) => {
+            setFilterData({ ...filterData, name: v });
+          }}
+          onClear={() => {
+            clearField('name');
+          }}
+          value={filterData['name']}
```

Comment:
value, onChange, placeholder, onClear should be on `AddFilterPattern.Search `, I think

Reply:
Fixed